src/main/java/org/opensearch/sdk/ExtensionRestHandler.java

```diff
@@ -19,12 +19,22 @@
  * This interface defines methods which an extension REST handler (action) must provide.
  * It is the Extension counterpart to core OpenSearch {@link RestHandler}.
  */
-public interface ExtensionRestHandler {
+public abstract class ExtensionRestHandler {
+
+    private String actionPrefix;
+
+    public ExtensionRestHandler(String extensionId) {
```

Comment:
Not clear on the motivation for abstract class vs. interface but I presume it's the presence of this variable?  If so should this be a `protected` constructor?   Also I'd prefer to keep interface but add an intermediate "BaseExtensionRestHandler" class, possibly to add functionality such as suggested in #128.